### ISSUE TITLE
11.0.2 Lua Errors

### ISSUE THREAD

**Issue by JeeveStobs** [Backupiseasy/ThreatPlates#544]
Starting with the 11.0.2 release, the following lua errors are thrown upon logging onto a character:

```
Message: Interface/AddOns/TidyPlates_ThreatPlates/Init.lua:148: attempt to call field 'GetSpellInfo' (a nil value)
Time: Tue Aug 13 16:20:43 2024
Count: 1
Stack: Interface/AddOns/TidyPlates_ThreatPlates/Init.lua:148: attempt to call field 'GetSpellInfo' (a nil value)
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Init.lua"]:148: in function <Interface/AddOns/TidyPlates_ThreatPlates/Init.lua:146>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Constants.lua"]:512: in function `InitializeTotemInformation'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Constants.lua"]:538: in function `GetDefaultTotemSettings'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Constants.lua"]:1713: in main chunk

Locals: (*temporary) = nil
(*temporary) = 2484
(*temporary) = "attempt to call field 'GetSpellInfo' (a nil value)"
```

```
Message: Interface/AddOns/TidyPlates_ThreatPlates/Init.lua:223: attempt to index field 'HeadlineView' (a nil value)
Time: Tue Aug 13 16:20:43 2024
Count: 1
Stack: Interface/AddOns/TidyPlates_ThreatPlates/Init.lua:223: attempt to index field 'HeadlineView' (a nil value)
[string "@Interface/AddOns/Details/Libs/AceAddon-3.0/AceAddon-3.0.lua"]:494: in function `InitializeAddon'
[string "@Interface/AddOns/Details/Libs/AceAddon-3.0/AceAddon-3.0.lua"]:619: in function <...ce/AddOns/Details/Libs/AceAddon-3.0/AceAddon-3.0.lua:611>

Locals: self = <table> {
 initializequeue = <table> {
 }
 statuses = <table> {
 }
 embeds = <table> {
 }
 frame = AceAddon30Frame {
 }
 addons = <table> {
 }
 enablequeue = <table> {
 }
}
addon = <table> {
 modules = <table> {
 }
 defaultModuleLibraries = <table> {
 }
 name = "TidyPlatesThreat"
 orderedModules = <table> {
 }
 baseName = "TidyPlates_ThreatPlates"
 enabledState = true
 defaultModuleState = true
}
```

```
Message: ...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:2173: attempt to index local 'db' (a nil value)
Time: Tue Aug 13 16:20:43 2024
Count: 2
Stack: ...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:2173: attempt to index local 'db' (a nil value)
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:2173: in function `UIScaleChanged'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:1792: in function `?'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:1302: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:1301>

Locals: self = <table> {
 IS_WRATH_CLASSIC = false
 Theme = <table> {
 }
 TotemCreatureType = "Totem"
 AurasFilterMode = <table> {
 }
 IS_CATA_CLASSIC = false
 ANCHOR_POINT_SETPOINT = <table> {
 }
 CVars = <table> {
 }
 THREAT_VALUE_TYPE = <table> {
 }
 Font = <table> {
 }
 WOW_USES_CLASSIC_NAMEPLATES = false
 IGNORED_STYLES = <table> {
 }
 PlatesByUnit = <table> {
 }
 Animations = <table> {
 }
 ThreatPlates = <table> {
 }
 ANCHOR_POINT_TEXT = <table> {
 }
 IS_TBC_CLASSIC = false
 MODE_FOR_STYLE = <table> {
 }
 TotemInformation = <table> {
 }
 ON_UPDATE_PER_FRAME = 0.032933
 CUSTOM_PLATES_GLOW_FRAMES = <table> {
 }
 Debug = <table> {
 }
 PlatesByGUID = <table> {
 }
 EventHandler = <table> {
 }
 LibUTF8 = <table> {
 }
 Logging = <table> {
 }
 PLATE_FADE_IN_TIME = 0.500000
 ANCHOR_POINT = <table> {
 }
 IS_TWW = false
 PlayerName = "Luxør"
 TOTEMS = <table> {
 }
 SCRIPT_FUNCTIONS = <table> {
 }
 ON_UPDATE_INTERVAL = 0.250000
 Data = <table> {
 }
 LibAceConfigRegistry = <table> {
 }
 LibAceConfigDialog = <table> {
 }
 db = <table> {
 }
 IS_CLASSIC = false
 WIDGET_EVENTS = <table> {
 }
 IS_CLASSIC_SOD = false
 Widgets = <table> {
 }
 CUSTOM_GLOW_WRAPPER_FUNCTIONS = <table> {
 }
 CUSTOM_GLOW_FUNCTIONS = <table> {
 }
 Cache = <table> {
 }
 TARGET_TEXTURES = <table> {
 }
 CASTBAR_INTERRUPT_HOLD_TIME = 1
 IS_MAINLINE = true
 LibCustomGlow = <table> {
 }
 DEFAULT_FONT = "Cabin"
 UIScale = 1
 DEBUG = true
 ADDON_DIRECTORY = "Interface\AddOns\TidyPlates_ThreatPlates\"
 BackdropTemplate = "BackdropTemplate"
 LEGACY_CUSTOM_NAMEPLATES = <table> {
 }
 LibSharedMedia = <table> {
 }
 PlatesCreated = <table> {
 }
 DEFAULT_SMALL_FONT = "Arial Narrow"
 GLOW_TYPES = <table> {
 }
 PlayerClass = "MAGE"
 IGNORED_STYLES_WITH_NAMEMODE = <table> {
 }
 PlatesVisible = <table> {
 }
}
db = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'db' (a nil value)"
Addon = <table> {
 IS_WRATH_CLASSIC = false
 Theme = <table> {
 }
 TotemCreatureType = "Totem"
 AurasFilterMode = <table> {
 }
 IS_CATA_CLASSIC = false
 ANCHOR_POINT_SETPOINT = <table> {
 }
 CVars = <table> {
 }
 THREAT_VALUE_TYPE = <table> {
 }
 Font = <table> {
 }
 WOW_USES_CLASSIC_NAMEPLATES = false
 IGNORED_STYLES = <table> {
 }
 PlatesByUnit = <table> {
 }
 Animations = <table> {
 }
 ThreatPlates = <table> {
 }
 ANCHOR_POINT_TEXT = <table> {
 }
 IS_TBC_CLASSIC = false
 MODE_FOR_STYLE = <table> {
 }
 TotemInformation = <table> {
 }
 ON_UPDATE_PER_FRAME = 0.032933
 CUSTOM_PLATES_GLOW_FRAMES = <table> {
 }
 Debug = <table> {
 }
 PlatesByGUID = <table> {
 }
 EventHandler = <table> {
 }
 LibUTF8 = <table> {
 }
 Logging = <table> {
 }
 PLATE_FADE_IN_TIME = 0.500000
 ANCHOR_POINT = <table> {
 }
 IS_TWW = false
 PlayerName = "Luxør"
 TOTEMS = <table> {
 }
 SCRIPT_FUNCTIONS = <table> {
 }
 ON_UPDATE_INTERVAL = 0.250000
 Data = <table> {
 }
 LibAceConfigRegistry = <table> {
 }
 LibAceConfigDialog = <table> {
 }
 db = <table> {
 }
 IS_CLASSIC = false
 WIDGET_EVENTS = <table> {
 }
 IS_CLASSIC_SOD = false
 Widgets = <table> {
 }
 CUSTOM_GLOW_WRAPPER_FUNCTIONS = <table> {
 }
 CUSTOM_GLOW_FUNCTIONS = <table> {
 }
 Cache = <table> {
 }
 TARGET_TEXTURES = <table> {
 }
 CASTBAR_INTERRUPT_HOLD_TIME = 1
 IS_MAINLINE = true
 LibCustomGlow = <table> {
 }
 DEFAULT_FONT = "Cabin"
 UIScale = 1
 DEBUG = true
 AD
```

```
Message: Interface/AddOns/TidyPlates_ThreatPlates/Core.lua:410: attempt to index field 'PersonalNameplate' (a nil value)
Time: Tue Aug 13 16:20:49 2024
Count: 1
Stack: Interface/AddOns/TidyPlates_ThreatPlates/Core.lua:410: attempt to index field 'PersonalNameplate' (a nil value)
[string "@Interface/AddOns/Blizzard_ExpansionLandingPage/Blizzard_ExpansionLandingPage.lua"]:78: in function `RefreshExpansionOverlay'
[string "@Interface/AddOns/Blizzard_ExpansionLandingPage/Blizzard_ExpansionLandingPage.lua"]:37: in function <...pansionLandingPage/Blizzard_ExpansionLandingPage.lua:35>

Locals: self = ExpansionLandingPage {
 overlay = <table> {
 }
 Overlay = Frame {
 }
}
newestOverlay = <table> {
}
```

**Comment by karpana** [comment]
quite likely a duplicate of #543

**Comment by JeeveStobs** [comment]
> quite likely a duplicate of #543

Probably but I documented the issue better with a descriptive title and used proper code formatting.

**Comment by Backupiseasy** [comment]
This issue should be fixed now with the newest update 11.0.4.

**Comment by F-Lambda** [comment]
Seems like there's still some bugs / lua errors with 12.0.5, albeit fewer than before:

```
Message: ...ns/TidyPlates_ThreatPlates/Widgets/StealthWidget.lua:152: attempt to call local 'UnitBuff' (a nil value)
Time: Wed Aug 14 08:05:59 2024
Count: 29
Stack: ...ns/TidyPlates_ThreatPlates/Widgets/StealthWidget.lua:152: attempt to call local 'UnitBuff' (a nil value)
[string "=[C]"]: in function `UnitBuff'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Widgets/StealthWidget.lua"]:152: in function `OnUnitAdded'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/Widgets/WidgetHandler.lua"]:387: in function `OnUnitAdded'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:586: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:572>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:627: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:603>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:672: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:645>
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:1381: in function `?'
[string "@Interface/AddOns/TidyPlates_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua"]:1302: in function <...s_ThreatPlates/TidyPlatesInternal/TidyPlatesCore.lua:1301>

Locals: 
```

**Comment by Backupiseasy** [comment]
Ok, I need to check more thoroghly where UnitBuff is still used as it was removed with 11.0.2. Though, no one really uses stealth widget, I think. It's not up to date anyway. I probably should remove it.

This Lua error should now be fixed with 12.0.6. It should be the last one :D
